src/store/AuthStore.js

```diff
@@ -0,0 +1,52 @@
+import { flow, types } from 'mobx-state-tree'
+import oauth from 'panoptes-client/lib/oauth'
+
+import { ASYNC_STATES } from '@util'
+
+const AuthStore = types.model('AuthStore', {
+  
+  status: types.optional(types.string, ASYNC_STATES.IDLE),
+  user: types.optional(types.frozen({}), null),  // When uninitialised, user should be null instead of {}
+  
+}).actions(self => ({
+  
+  initialise () {
+    self.checkUser()
+  },
+
+  checkUser: flow(function * checkUser () {
+    self.status = ASYNC_STATES.FETCHING
+    oauth.checkCurrent()
+    .then(user => {
+      self.status = ASYNC_STATES.SUCCESS
+      self.user = user
+    })
+  }),
+
+  login () {
+    oauth.signIn(computeRedirectURL(window))
+  },
+
+  logout: flow(function* () {
```

Comment:
I think you need to name the generator here.
```suggestion
  logout: flow(function * logout() {
```

Reply:
At least, when I started working with generators, they wouldn't work with anonymous functions.

Reply:
Thanks! I've just updated the code to add a name to the anon logout function, though I'm still encountering that darn `regenerator undefined` error.

Reply:
Wait I didn't realise in GitHub you can make code changes directly in your comments, and that I can commit the suggestions from within the PR's comments. Sweet!